### PR TITLE
Bs 598 mapping correction

### PR DESCRIPTION
**Pull request by msatta**

### CODE REVIEW

File: app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala [hunk 1]
```diff
@@ -69,29 +68,22 @@ trait UpstreamConnector extends HttpErrorFunctions with Logging with UsingCircui
...
     implicit r: RequestId
-  ): ResponseValidation[T] =
-    statusCode match {
-      case BAD_GATEWAY => logAndGenDownstreamResponse(error, BAD_GATEWAY, message, UPSTREAM_BAD_GATEWAY)
-      case SERVICE_UNAVAILABLE =>
-        logAndGenDownstreamResponse(error, SERVICE_UNAVAILABLE, message, UPSTREAM_SERVICE_UNAVAILABLE)
-      case GATEWAY_TIMEOUT => logAndGenDownstreamResponse(error, GATEWAY_TIMEOUT, message, UPSTREAM_TIMEOUT)
-      case _ => logAndGenDownstreamResponse(error, SERVICE_UNAVAILABLE, message, UPSTREAM_SERVICE_UNAVAILABLE)
-    }
+  ): ResponseValidation[T] = logAndGenDownstreamResponse(error, SERVICE_UNAVAILABLE, message, BaseError.SERVER_ERROR)
```
Comment (kurtlogan) [inline app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala]: This isn't mentioned in the ticket, I assume you are wanitng to return ServerError for all upstream 5xx errors. I noticed a comment on the ticket saying you were going to map gateway timeout to server error and I assume you are wanting to map all upstream errors to server error, which is good practice so we don't leak internal issues to thrid parties. If this is the case could you update the ticket so we have a record of it

also don't use SERVICE_UNAVAILABLE for status code, this value is used in logging so we want the actual status code that the upstream service has thrown - see suggestion
```suggestion
  ): ResponseValidation[T] = logAndGenDownstreamResponse(error, statusCode, message, BaseError.SERVER_ERROR)
```

**Review by kurtlogan — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala (hunk 1 above)
Comment (msatta) [reply app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala]: Yes, you're right. it's the desired behavior. I'll update the ticket.
About the SERVICE_UNAVAILABLE (503) as status code, it's the same the Platform API is returning. And the platform overrides the response anyway. At the moment we are keeping a 1-to-1 mapping with their responses. Should we change it? If so, we need to do some work in the acceptance test as well.

Comment (kurtlogan) [reply app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala]: Yeah but that status code is only used for logging. The value we are returning to the client is `BaseError.SERVER_ERROR`
See https://github.com/hmrc/breathing-space-if-proxy/blob/main/app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala#L120

Comment (msatta) [reply app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala]: Right! The ticket has been updated and the commit suggestion applied

**Review by kurtlogan — APPROVED** [review]

**Pull request merged**